### PR TITLE
SSHOperator - Restore ability to override SSHHook cmd_timeout

### LINKED ISSUE #30167: SSHOperator - Allow specific command timeout
### Description

Following #29282, command timeout is set at the `SSHHook` level while it used to be able to set at the `SSHOperator` level.

I will work on a PR as soon as i can.

### Use case/motivation

Ideally, i think we could have a default value set on `SSHHook`, but with the possibility of overriding it at the `SSHOperator` level.

### Related issues

#29282 

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by punx120**
Fixes #30167 .

@Aakcht the only way to disable timeout is to do it at the `SSHHook` level as shown in one of the test. Hopefully that's enough for you.

### CODE REVIEW

File: tests/providers/ssh/hooks/test_ssh.py [hunk 1]
```diff
@@ -1028,6 +1010,24 @@ def test_command_timeout_fail(self):
                     None,
                 )
 
+    def test_command_timeout_not_set(self):
```
Comment (punx120) [inline tests/providers/ssh/hooks/test_ssh.py:1013]: this illustrates how to set _infinite_ timeout

File: tests/providers/ssh/hooks/test_ssh.py [hunk 2]
```diff
@@ -974,24 +974,6 @@ def test_exec_ssh_client_command(self):
             )
             assert ret == (0, b"airflow\n", b"")
 
-    @pytest.mark.flaky(reruns=5)
```
Comment (punx120) [inline tests/providers/ssh/hooks/test_ssh.py:977]: Deleted as it's basically the same test as `test_command_timeout_fail`

File: tests/providers/ssh/hooks/test_ssh.py (hunk 1 above)
Comment (Aakcht) [reply tests/providers/ssh/hooks/test_ssh.py:1013]: Looks like with your changes when `cmd_timeout` is not set in hook/operator, but set to `null` in connection extra it still means "infinite timeout", so for me it's ok.

**Review by Aakcht — APPROVED** [review]

File: airflow/providers/ssh/hooks/ssh.py [hunk 3]
```diff
@@ -471,11 +471,11 @@ def exec_ssh_client_command(
         command: str,
         get_pty: bool,
         environment: dict | None,
-        timeout: int | None = None,
+        timeout: int | ArgNotSet | None = None,
```
Comment (uranusjr) [inline airflow/providers/ssh/hooks/ssh.py]: I feel the default be changed to `NOTSET`. And the `None` case below should be dropped entirely…?

Comment (punx120) [reply airflow/providers/ssh/hooks/ssh.py]: yeah that seems reasonable. Let me me take a look

File: tests/providers/ssh/hooks/test_ssh.py (hunk 1 above)
Comment (punx120) [reply tests/providers/ssh/hooks/test_ssh.py:1013]: correct, you can set it to `null` in connection extra or `None` when you create the hook. Also i think after we replace the default value from `None` to `ArgNotSet` in `SSHOperator` as suggested by @uranusjr, we should also be able to set it as the operator level.

File: airflow/providers/ssh/hooks/ssh.py (hunk 3 above)
Comment (punx120) [reply airflow/providers/ssh/hooks/ssh.py]: updated - much better imho. ty

**Review by uranusjr — APPROVED** [review]

**Comment by uranusjr** [comment]
I added a commit to fix static checks; the logic shouldn’t change with it.

**Review by potiuk — APPROVED** [review]

**Pull request merged**